versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ExportDatabaseAdapter.java

```diff
@@ -97,20 +99,25 @@ private HeadsAndForkPoints scanAllReferences(Consumer<CommitLogEntry> commitHand
           .map(ReferenceInfo::getHash)
           .forEach(
               head -> {
-                try (Stream<CommitLogEntry> commits = databaseAdapter.commitLog(head)) {
-                  for (Iterator<CommitLogEntry> commitIter = commits.iterator();
-                      commitIter.hasNext(); ) {
-                    CommitLogEntry commit = commitIter.next();
-                    if (!identify.handleCommit(commit)) {
-                      break;
-                    }
-                    commitHandler.accept(commit);
+                Deque<Hash> commitsToProcess = new ArrayDeque<>();
+                commitsToProcess.push(head);
+                while (!commitsToProcess.isEmpty()) {
+                  Hash hash = commitsToProcess.pop();
+                  try (Stream<CommitLogEntry> commits = databaseAdapter.commitLog(hash)) {
+                    commits.forEach(
+                        commit -> {
+                          if (identify.handleCommit(commit)) {
+                            commitHandler.accept(commit);
+                            for (Hash h : commit.getAdditionalParents()) {
+                              commitsToProcess.push(h);
```

Comment:
Why not append? If we have two parallel branches with cross-merges processing newer commits first will reduce the number of commit log calls, I guess. If branches are not related, then there is perhaps no difference.